guacamole/src/main/java/org/apache/guacamole/rest/auth/AuthenticationService.java

```diff
@@ -392,7 +333,15 @@ private List<DecoratedUserContext> getUserContexts(GuacamoleSession existingSess
 
                 // Update existing UserContext
                 AuthenticationProvider authProvider = oldUserContext.getAuthenticationProvider();
-                UserContext updatedUserContext = authProvider.updateUserContext(oldUserContext, authenticatedUser, credentials);
+                UserContext updatedUserContext;
+                try {
+                    updatedUserContext = authProvider.updateUserContext(oldUserContext, authenticatedUser, credentials);
+                }
+                catch (GuacamoleException | RuntimeException | Error e) {
```

Comment:
This feels a little overly broad to be catching `RuntimeException`s and `Error`s. Why catch those now?

Reply:
At the authorization stage, any error whatsoever from any extension will abort the authorization process and fail overall authentication. If this `catch` were not broad enough to also include unchecked exceptions/errors, then those errors would not be listenable via `Listener` and `FailureEvent`.

Reply:
Makes sense.